polkadot/runtime/common/src/identity_migrator.rs

```diff
@@ -112,20 +113,31 @@ pub mod pallet {
 		))]
 		pub fn reap_identity(
 			origin: OriginFor<T>,
-			who: T::AccountId,
+			who_vec: Vec<T::AccountId>,
 		) -> DispatchResultWithPostInfo {
 			T::Reaper::ensure_origin(origin)?;
 			// - number of registrars (required to calculate weight)
 			// - byte size of `IdentityInfo` (required to calculate remote deposit)
 			// - number of sub accounts (required to calculate both weight and remote deposit)
-			let (registrars, bytes, subs) = pallet_identity::Pallet::<T>::reap_identity(&who)?;
-			T::ReapIdentityHandler::on_reap_identity(&who, bytes, subs)?;
-			Self::deposit_event(Event::IdentityReaped { who });
+			let tuple_vec: Vec<(T::AccountId, u32, u32, u32)> =
+				pallet_identity::Pallet::<T>::reap_identity(who_vec.clone())?;
+
+			let mut total_registrars: u32 = 0;
+			let mut total_subs: u32 = 0;
+			for (who, registrars, bytes, subs) in tuple_vec {
+				T::ReapIdentityHandler::on_reap_identity(&who, bytes, subs)?;
+
+				total_registrars = total_registrars + registrars;
+				total_subs = total_subs + subs;
+			}
+
+			Self::deposit_event(Event::IdentityReaped { who_vec });
 			let post = PostDispatchInfo {
 				actual_weight: Some(<T as pallet::Config>::WeightInfo::reap_identity(
-					registrars, subs,
+					total_registrars,
+					total_subs,
 				)),
-				pays_fee: Pays::No,
+				pays_fee: Pays::Yes,
```

Comment:
This is not really correct, as we want to omit fee payment in the success case.  
Normally we go for something like >=90% of good values justifies a free transaction.

Reply:
how do i identify good values to use as a justification of free transaction?